crates/tabby-common/src/config.rs

```diff
@@ -145,6 +146,13 @@ impl Default for ServerConfig {
 #[async_trait]
 pub trait RepositoryAccess: Send + Sync {
     async fn list_repositories(&self) -> Result<Vec<RepositoryConfig>>;
+
+    fn clear_index(&self) -> Result<()> {
+        Ok(())
+    }
+    fn write_index(&self, _source_file: SourceFile) -> Result<()> {
+        Ok(())
+    }
```

Comment:
```suggestion
    fn start_snapshot(&self, version: u64) {}
    fn process_file(&self, version: u64, file: SourceFile) {}
    fn end_snapshot(&self, version: u64) {}
```

Each returns void - ensuring the error is handled internally in implementation.